train.py

```diff
@@ -932,14 +968,15 @@ def train_one_epoch(
                         normalize=True
                     )
 
-        if saver is not None and args.recovery_interval and (
-                last_batch or (batch_idx + 1) % args.recovery_interval == 0):
-            saver.save_recovery(epoch, batch_idx=batch_idx)
-
-        if lr_scheduler is not None:
-            lr_scheduler.step_update(num_updates=num_updates, metric=losses_m.avg)
+        if need_step:
+            if saver is not None and args.recovery_interval and (
```

Comment:
I'd put this along with the block above under one `if need_step:` , and have the block above in it's own `if (batch_idx // args.iters_to_accum) % args.log_interval == 0` indent

Reply:
I'll try to modify the block as you mentioned.
Thanks.